Getting-Started-with-Akka-http-signature.md

```diff
@@ -0,0 +1,304 @@
+"""
+``` 
+This will return a String representation of the Pointed Graph in turtle format. This representation can also be saved on your local file system like so:
+
+```scala
+write(wd/"publicKey.ttl", RSAKeys.save(toTurtle))
+```
+
+----------------------------
+
+### Attaching Public keys to a File/URI
+
+One can use the `cp` or the `mv` Ammonite commands to move the public key file into the test_www directory which resides within the rww-play directory. The process of attaching the file, containing the key to a URI is very similar.
+
+```scala
+cp(wd/"publicKey.ttl" , wd/"rww-play"/"test_www"/"pubKey.ttl")
+```
+
+The Rww-play web server, which the Akka-http-signature library makes use of, uses symbolic links to indicate default representations of different documents. The Http content negotiation functionality is thus preserved. Http Content negotiation is a mechanism that is used for representing different representations of a resource at the same URI. The user can then specify which is best suited for them. 
+
+Content negotiation is preserved because the user can link to files without specifically addressing them via their original name by using these symbolic links. This makes it possible for links to be followed both on the file system and on the web in the same way. 
+
+This can be quite a complex mechanism and the setup for it can vary greatly between different servers.
+
+To create a symbolic link, one can make a symbolic link to their public key file using the following commands in the bash console:
+
+```bash
+$ ln -s pubKey.ttl pubKey
+```
+### Access Control
+
+Currently in order to manipulate the access control one can use `curl` and `PATCH` commands [As detailed in the rww-play wiki page](https://github.com/read-write-web/rww-play/wiki/Curl-Interactions).
+
+To demonstrate how access control changes user permitted actions, one can also manually edit the .acl.ttl file. To do this, one must navigate to the test_www directory (or the server that contains the Key files) and look for the correct file. After that in order to permit access to the , the user must comment out the following:
+
+```bash
+@prefix acl: <http://www.w3.org/ns/auth/acl#> . 
+@prefix cert: <http://www.w3.org/ns/auth/cert#> .
+
+
+[] acl:accessTo <pubKey>, <pubKey.ttl> ;
+
+   acl:agent <pubKey#me> ;
+
+#   acl:agent [ cert:key <publicKey#key> ];
+
+	acl:agentClass <http://xmlns.com/foaf/0.1/Agent> ;
+
+	acl:mode acl:Read .
+
+
+
+<> acl:include <.acl> .
+```
+
+This allows the user to manipulate access control by simply commenting out `acl:agent [ cert:key <publicKey#key> ];` 
+
+After that, the access to the Key files specified will be restricted. And therefore they will not be usable via the symbolic link created earlier. 
+
+If one attempts to do so, via curl for example, they will be presented with the following error message: 
+
+```bash
+$ curl -i -k https://localhost:8443/2013/pubKey.acl
+
+curl: (7) Couldn't connect to server
```

Comment:
Try first to connect to `https://localhost:8443/2012/card` since that is the document whose acl we have changed, to see that it has. This may actually require restarting the server - it should not, but that's a bug.